evaluator/builtin_time_test.go

```diff
@@ -217,13 +220,11 @@ func (s *testEvaluatorSuite) TestDateFormat(c *C) {
 			v.GetValue(), t["Expect"][0].GetValue()))
 	}
 
-	// error
 	ds := types.MakeDatums("0000-01-00 00:00:00.123456",
 		"%b %M %m %c %D %d %e %j %k %h %i %p %r %T %s %f %U %u %V %v %a %W %w %X %x %Y %y %%")
-	_, err := builtinDateFormat(ds, nil)
-	// Some like dayofweek() doesn't support the date format like 2000-00-00 returns 0,
-	// so it returns an error.
-	c.Assert(err, NotNil)
+	v, err := builtinDateFormat(ds, nil)
+	c.Assert(err, IsNil)
+	c.Assert(v, testutil.DatumEquals, types.Datum{})
```

Comment:
The result of MySQL is 'Jan January 01 1 0th 00 0 000 0 12 00 AM 12:00:00 AM 00:00:00 00 123456 52 00 52 52 Sat Saturday 6 4294967295 4294967295 0000 00 % '